app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -86,9 +86,7 @@ class WifiDirectManager @Inject constructor(
   // Corresponds to P2P group formed between the two devices
   private lateinit var groupInfo: WifiP2pInfo
```

Comment:
I only mentioned this to you in chat but this property can be removed I think, instead of assigning it just pass it through methods and things should start to make a lot more sense I think

Reply:
@macgills Android studio shows errors if I remove it.It suggests to create an abstract property or local variable

Reply:
show me a screenshot of the error and we can walk through it

Reply:
<img width="709" alt="Screenshot 2020-11-11 at 3 37 51 PM" src="https://user-images.githubusercontent.com/43576162/98798408-ddea8b80-2433-11eb-9682-81a7de03a7b0.png">

Reply:
> instead of assigning it just pass it through methods 

so delete isGroupFormed, isGroupOwner and groupOwnerAddress and do `performHandshakeWithSelectedPeerDevice(groupInfo)` you could even rename this to `performHandshakeWith(groupInfo)` and it reads a lot better

>things should start to make a lot more sense I think

Reply:
so how can we start file transfer if we don't know `groupformed` ` groupowner` and `groupownerAddress`?

Reply:
pass in the groupInfo, the groupInfo knows these things, not the wifiDirectManager

Reply:
try `.handshake(groupInfo)`

Reply:
no i mean here
```java
private fun startFileTransfer() {
    if (isGroupFormed) {
      if (isFileSender) {
        Log.d(LocalFileTransferFragment.TAG, "Starting file transfer")
        fileReceiverDeviceAddress =
          if (isGroupOwner) selectedPeerDeviceInetAddress else groupOwnerAddress
        activity.toast(R.string.preparing_files, Toast.LENGTH_LONG)
        val senderDevice = SenderDevice(this, activity)
        lifecycleCoroutineScope.launch {
          val isFileSendSuccessfully = senderDevice.send(filesForTransfer)
          onFileTransferAsyncTaskComplete(isFileSendSuccessfully)
          if (BuildConfig.DEBUG) {
            Log.d(TAG, "SenderDevice completed $isFileSendSuccessfully")
          }
        }
      } else {
        callbacks?.onFilesForTransferAvailable(filesForTransfer)
        val receiverDevice = ReceiverDevice(this)
        lifecycleCoroutineScope.launch {
          val isReceivedFileSuccessFully = receiverDevice.receive()
          onFileTransferAsyncTaskComplete(isReceivedFileSuccessFully)
          if (BuildConfig.DEBUG) {
            Log.d(TAG, "ReceiverDevice completed $isReceivedFileSuccessFully")
          }
        }
      }
    }
  }
```

Reply:
```
  fun setClientAddress(clientAddress: InetAddress) {

    // If control reaches here, means handshake was successful
    selectedPeerDeviceInetAddress = clientAddress
    startFileTransfer()
  }
```

this method is probably stupid, also misleading delete it and just call `startFileTransfer(clientAddress, groupInfo)`

Reply:
`selectedPeerDeviceInetAddress ` this means this is probably also not needed and can be deleted